mlflow/utils/autologging_utils/logging_and_warnings.py

```diff
@@ -50,8 +51,14 @@ def _patched_showwarning(self, message, category, filename, lineno, *args, **kwa
 
         # If the warning's source file is contained within the MLflow package's base
         # directory, it is an MLflow warning and should be emitted via `logger.warning`
-        warning_source_path = Path(filename).resolve()
-        is_mlflow_warning = self._mlflow_root_path in warning_source_path.parents
+        # NB: converting the filename to a string to avoid a potential recursion issue
+        # if the filename being passed is a Path type object.
+        try:
+            warning_source_path = Path(str(filename)).resolve()
+            is_mlflow_warning = self._mlflow_root_path in warning_source_path.parents
+        except Exception:
+            is_mlflow_warning = False
```

Comment:
Do we need this change? Recursion can occur in other places.

Reply:
I forgot to add that commit ;)